src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -433,5 +449,52 @@ void WiFiManager::PostConnectivityStatusChange(ConnectivityChange changeType)
     PlatformMgr().PostEventOrDie(&networkEvent);
 }
 
+System::Clock::Milliseconds32 WiFiManager::GetNextRecoveryTime()
+{
+
+    if (mConnectionRecoveryTimeMs >= kConnectionRecoveryMaxIntervalMs)
+    {
+        // reset previous connection recovery time to max value
+        mConnectionRecoveryTimeMs = kConnectionRecoveryMaxIntervalMs;
+        // find the new random jitter value in range [-jitter, +jitter]
+        int32_t jitter            = chip::Crypto::GetRandU32() % (2 * jitter + 1) - jitter;
+        mConnectionRecoveryTimeMs = mConnectionRecoveryTimeMs + jitter;
+    }
+    else
+    {
+        mConnectionRecoveryTimeMs = mConnectionRecoveryTimeMs * 2;
+    }
+    return System::Clock::Milliseconds32(mConnectionRecoveryTimeMs);
+}
+
+void WiFiManager::Recover(System::Layer * layer, void * param)
+{
+    // Prevent executing Scan if it has been aborted by the failsafe timer occurrence.
+    if (!Instance().mRecoveryTimerAborted)
+    {
+        ChipLogProgress(DeviceLayer, "Connection recover re-scanning... (next attempt in %d ms)",
+                        Instance().GetNextRecoveryTime().count());
+        Instance().mInternalScan = true;
```

Comment:
Please remove this line, as the `mInternalScan` will be set by the Scan() call below.